daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -2135,6 +2143,39 @@ case class FNTwoArgsMathExpr(nameAsParsed: String, fnQName: RefQName,
   }
 }
 
+case class DFDLXShiftExpr(nameAsParsed: String, fnQName: RefQName,
+                             args: List[Expression], constructor: (List[CompiledDPath], NodeInfo.Kind) => RecipeOp)
+  extends FunctionCallBase(nameAsParsed, fnQName, args) {
+  checkArgCount(2)
```

Comment:
Move this checkArgCount(2) inside the compiledDPath method to be consistent with other functions. It should be the first line in that method. 

We tend to do checks inside methods, rather than at object construction time, because the constructed object often is the source of contextual information that is used to create a good diagnostic message; hence, fully constructing the object before doing anything that could detect an error is preferred.

This is the daffodil version of the common Java convention/principle to never do anything that could throw an exception in a constructor.

Reply:
checkArgCount(2) was added to comileDPath, but was not removed here.